pilot/pkg/model/sidecar.go

```diff
@@ -403,16 +410,19 @@ func ConvertToSidecarScope(ps *PushContext, sidecarConfig *config.Config, config
 	// that these services need
 	out.servicesByHostname = make(map[host.Name]*Service, len(out.services))
 	out.destinationRules = make(map[host.Name]*config.Config)
+	out.destinationRulesByNames = make(map[ConfigKey]*config.Config)
 	for _, s := range out.services {
 		out.servicesByHostname[s.Hostname] = s
 		dr := ps.DestinationRule(&dummyNode, s)
 		if dr != nil {
-			out.destinationRules[s.Hostname] = dr
-			out.AddConfigDependencies(ConfigKey{
+			ckey := ConfigKey{
 				Kind:      gvk.DestinationRule,
 				Name:      dr.Name,
 				Namespace: dr.Namespace,
-			})
+			}
+			out.destinationRules[s.Hostname] = dr
+			out.destinationRulesByNames[ckey] = dr
```

Comment:
This is not right, In L416 `dr := ps.DestinationRule(&dummyNode, s)`, the destinationrule got is merged, so the name only include one, but i think here you want all the original names.

Reply:
could you point me to where the merging is happening?

Reply:
I think i answered above. It is mergeDestinationRule, ptal

Reply:
+1, seems like merging is not accounted for here